core/src/main/scala/kafka/log/TimeIndex.scala

```diff
@@ -55,17 +55,21 @@ class TimeIndex(_file: File, baseOffset: Long, maxIndexSize: Int = -1, writable:
 
   override def entrySize = 12
 
+  @volatile var _lastEntry = lastEntryFromIndexFile
```

Comment:
This should be `private` since that's the reason for having an accessor. Also, I'd have it at the top of the file instead of after `entrySize` (method ordering doesn't matter, but val/var ordering does matter).

Reply:
@ijuma Thanks for the quick review. Yeah I forgot to make this private. I have updated the patch as suggested.